Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrTextEmitter.kt

```diff
@@ -0,0 +1,670 @@
+        return "${renderStubType(superClassInit.type)}$parameters"
+    }
+
+    private fun renderStubType(stubType: StubType): String = when (stubType) {
+        is WrapperStubType -> stubType.kotlinType.render(kotlinFile)
+        is ClassifierStubType -> kotlinFile.reference(stubType.classifier) + renderTypeArguments(stubType.typeArguments)
+        is SymbolicStubType -> stubType.name + if (stubType.nullable) "?" else ""
+    }
+
+    private fun renderValueUsage(value: ValueStub): String = when (value) {
+        is StringConstantStub -> value.value
+        is IntegralConstantStub -> renderIntegralConstant(value)!!
+        is DoubleConstantStub -> renderDoubleConstant(value)!!
+        is ConstructorParamStub -> value.name
+    }
+
+    private fun renderAnnotation(annotationStub: AnnotationStub): String = when (annotationStub) {
+        AnnotationStub.ObjC.ConsumesReceiver -> "@CCall.ConsumesReceiver"
+        AnnotationStub.ObjC.ReturnsRetained -> "@CCall.ReturnsRetained"
+        is AnnotationStub.ObjC.Method -> {
+            val stret = if (annotationStub.isStret) ", true" else ""
+            val selector = annotationStub.selector.quoteAsKotlinLiteral()
+            val encoding = annotationStub.encoding.quoteAsKotlinLiteral()
+            "@ObjCMethod($selector, $encoding$stret)"
+        }
+        is AnnotationStub.ObjC.Factory -> {
+            val stret = if (annotationStub.isStret) ", true" else ""
+            val selector = annotationStub.selector.quoteAsKotlinLiteral()
+            val encoding = annotationStub.encoding.quoteAsKotlinLiteral()
+            "@ObjCFactory($selector, $encoding$stret)"
+        }
+        AnnotationStub.ObjC.Consumed ->
+            "@CCall.Consumed"
+        is AnnotationStub.ObjC.Constructor ->
+            "@ObjCConstructor(${annotationStub.selector.quoteAsKotlinLiteral()}, ${annotationStub.designated})"
+        is AnnotationStub.ObjC.ExternalClass -> {
+            val protocolGetter = annotationStub.protocolGetter.quoteAsKotlinLiteral()
+            val binaryName = annotationStub.binaryName.quoteAsKotlinLiteral()
+            "@ExternalObjCClass" + when {
+                annotationStub.protocolGetter.isEmpty() && annotationStub.binaryName.isEmpty() -> ""
+                annotationStub.protocolGetter.isEmpty() -> "($binaryName)"
+                annotationStub.binaryName.isEmpty() -> "($protocolGetter)"
+                else -> "($protocolGetter, $binaryName)"
+            }
+        }
+        AnnotationStub.CCall.CString ->
+            "@CCall.CString"
+        AnnotationStub.CCall.WCString ->
+            "@CCall.WCString"
+        is AnnotationStub.CCall.Symbol ->
+            "@CCall(${annotationStub.symbolName.quoteAsKotlinLiteral()})"
+        is AnnotationStub.CStruct ->
+            "@CStruct(${annotationStub.struct.quoteAsKotlinLiteral()})"
+        is AnnotationStub.CNaturalStruct ->
+            "@CNaturalStruct(${annotationStub.struct.quoteAsKotlinLiteral()})"
```

Comment:
Doesn't look correct.
Please compare stubs generated for Kotlin/JVM too (e.g. libclang and LLVM bindings).